### DISCUSSION TITLE
Error loading ASGI app. Could not import module "main".

### DISCUSSION THREAD

**Discussion by misaka10843** [fastapi/fastapi#9838] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI
import comic.CreateComic
from Routers import ComicRouter
from database import Database

app = FastAPI()
db = Database()
```

### Description

After importing `comic.CreateComic`, the Error loading ASGI app appeared Could not import module 'main'
May I ask if the import method is incorrect?
I tried several different methods, but none of them worked

### Operating System

Windows

### Operating System Details

Windows 11 Pro 22624.1755

### FastAPI Version

0.100.0

### Python Version

Python 3.10.11

### Additional Context

![image](https://github.com/tiangolo/fastapi/assets/69132853/6e2d4e84-01ce-4478-9782-822eb344055c)

**Comment by mikeedjones** [comment]
What is `BaseModel.py` - there's an important Pydantic class with the same name and that could be causing some issues.

**Reply by misaka10843** [reply]
This is `BaseModel.py`
```python
from typing import List
from datetime import date

from pydantic import BaseModel


class ComicCreate(BaseModel):
    comic_name: str
    comic_date: date
    comic_intro: str
    comic_cover: str
    comic_magazine: str
    categories: List[str]
    tags: List[str]

```

After changing the name, I still couldn't solve the problem, so it should not be related to this

**Reply by mikeedjones** [reply]
oh - `from comic import ComicCreate` is the correct format. Might be good to get a [linter](https://pypi.org/project/pylint/) or a language server running.

**Reply by misaka10843** [reply]
I have tried `from comic import ComicCreate`, but it still prompts me to send the code and the detection results of pylint to ChatGPT, and then use the method provided by GPT to report the same error🤣

**Reply by misaka10843** [reply]
https://github.com/misaka10843/test-api
This is my repository

The abnormal situation is in the following two lines. After deleting these two lines, an error message will be reported and resolved

https://github.com/misaka10843/test-api/blob/865d305a19390ab7275bc03cacadb277e9d0bef0/Routers/ComicRouter.py#L2-L3

**Reply by mikeedjones** [reply]
pardon? By language server I mean [Pylance](https://marketplace.visualstudio.com/items?itemName=ms-python.vscode-pylance) or whatever the pycharm equivalent is.

**Reply by mikeedjones** [reply, marked as answer]
You have a circular import - line 5 in `CreateComic.py` imports from main`from main import db` - causing an error.

**Reply by misaka10843** [reply]
I see. I didn't check carefully🤣
Thank you very much for helping me find the error💕

**Reply by mikeedjones** [reply]
You're welcome. Seems like uvicorn dying quietly and not displaying the stack trace is an issue. Worth seeing if there's an open issue on the uvicron repo.

**Reply by Kludex** [reply]
No open issue there. You can open one with an MRE.

Uvicorn maintainer here.

**Reply by mikeedjones** [reply]
opened a discussion https://github.com/encode/uvicorn/discussions/2034

**Comment by estosito999** [comment]
![Captura de pantalla 2023-12-18 084827](https://github.com/tiangolo/fastapi/assets/136049915/8300c48d-a241-45d2-bbc8-b688e6e8fb32)
help mee I get the same error

**Reply by mikeedjones** [reply]
Please try `python env\main.py` - should produce a more helpful error message.

**Comment by Hiru1003** [comment]
thankyou
